### PR TITLE
created generic button component with its styles

### PR DESCRIPTION
**Pull request by Ran-Regev-HLS**

### CODE REVIEW

File: src/assets/components/shared/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  variant:
+    | "primary"
+    | "secondary"
+    | "secondaryGrey"
+    | "tertiary"
+    | "tertiaryGrey";
+  size: "large" | "medium" | "small";
+  isDisabled?: boolean;
+  onClick: () => void;
+  children?: React.ReactNode;
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:13]: import ReactNode from react

File: src/assets/components/shared/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { StyledButton } from "./styles";
+
+export interface ButtonProps {
+  variant:
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:4]: Create an enum 
```
export enum ButtonVariant {
PRIMARY="primary",
etc...
}
```

checkout raycast change case extension - you can do realy cool stuff when defining enums (Remind me tomorrow to show you how i would do that)

File: src/assets/components/shared/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    | "primary"
+    | "secondary"
+    | "secondaryGrey"
+    | "tertiary"
+    | "tertiaryGrey";
+  size: "large" | "medium" | "small";
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:10]: Same here - create an enum - ButtonSize

File: src/assets/components/shared/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import { StyledButton } from "./styles";
+
+export interface ButtonProps {
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:3]: What about start and end icon?

File: src/assets/components/shared/Button/Button.tsx (hunk 1 above)
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:13]: Do i have to send a react node to the button as children?
If i only want to show some text why would i have to wrap it in a reactNode?

File: src/assets/components/shared/Button/styles.ts [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    &:disabled {
+      background-color: transparent;
+      color: #c7cdd3;
+    }
+  `,
+};
+
+const sizeStyles = {
+  large: css`
+    width: Hug (80px) px;
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:112]: You are not supposed to use the Hug and Fixed properties from the figma - this is for the disgeners. I also asked it when I started working with figma

File: src/assets/components/shared/Button/styles.ts [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    height: Fixed (40px) px;
+    top: 140px;
+    left: 20px;
+    padding: 12px 16px 12px 16px;
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:116]: Create a spacing object in the same file you defined the colors object.
export const spacing = {
1: '4px',
2: '8px',
go on until 40px
}

File: src/assets/components/shared/Button/styles.ts [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+    gap: 4px;
+    border-radius: 4px 0px 0px 0px;
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:118]: Same comment as comment about spacing.
create border radius constant and use it

File: src/assets/components/shared/Button/styles.ts [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+};
+
+const sizeStyles = {
+  large: css`
+    width: Hug (80px) px;
+    height: Fixed (40px) px;
+    top: 140px;
+    left: 20px;
+    padding: 12px 16px 12px 16px;
+    gap: 4px;
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:117]: use spacing

File: src/assets/components/shared/Button/styles.ts [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    top: 20px;
+    left: 20px;
+    padding: 9px 16px 9px 16px;
+    gap: 4px;
+    border-radius: 4px 0px 0px 0px;
+    opacity: 0px;
+  `,
+};
+
+export const StyledButton = styled.button<ButtonProps>`
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:143]: Move this to the top of the file

File: src/assets/components/shared/Button/Button.tsx (hunk 4 above)
Comment (kfir-azu) [reply src/assets/components/shared/Button/Button.tsx:3]: I would also set an isLoading optional prop - if you pass isLoading render a loader on the button to indicate the user that something is happening behind the scenes

File: src/assets/components/shared/Button/Button.tsx [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+  children,
+}: ButtonProps) {
+  return (
+    <StyledButton
+      variant={variant}
+      size={size}
+      onClick={onClick}
+      disabled={isDisabled}
+    >
+      {children}
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/Button.tsx:30]: {isLoading ? Loader (checkout material ui loaders) : children}

File: src/assets/components/shared/Button/styles.ts (hunk 9 above)
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:143]: Create props for the styled component - not all button props are relevant to the styled component.
Maybe you can rely on the buttonProps and leave only the relevant fields

**Review by kfir-azu — CHANGES_REQUESTED** [review]

File: src/assets/components/shared/Button/Button.tsx (hunk 1 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:13]: done

File: src/assets/components/shared/Button/Button.tsx (hunk 2 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:4]: done

File: src/assets/components/shared/Button/Button.tsx (hunk 3 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:10]: done

File: src/assets/components/shared/Button/Button.tsx (hunk 4 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:3]: done

File: src/assets/components/shared/Button/Button.tsx (hunk 1 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:13]: done

File: src/assets/components/shared/Button/styles.ts (hunk 5 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:112]: done

File: src/assets/components/shared/Button/styles.ts (hunk 6 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:116]: done

File: src/assets/components/shared/Button/styles.ts (hunk 7 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:118]: done

File: src/assets/components/shared/Button/styles.ts (hunk 8 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:117]: done

File: src/assets/components/shared/Button/styles.ts (hunk 9 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:143]: done

Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/styles.ts:143]: done

File: src/assets/components/shared/Button/styles.ts [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+    }
+  `,
+};
+
+
+export function getCircularSize($size: ButtonSize):number {
+  return 0.8 * sizeDimensions[$size].height;
+}
+
+export const sizeDimensions = {
```
Comment (kfir-azu) [inline src/assets/components/shared/Button/styles.ts:141]: Change name to buttonSizeMap (use f2 to change across all app)

File: src/colors.ts [hunk 12]
```diff
@@ -23,3 +23,15 @@ export const colors = {
     _500: "#4C3D01",
   },
 };
+
+export const spacing = {
```
Comment (kfir-azu) [inline src/colors.ts:27]: Add values up to 40px, incrementing by 4px for each step

**Review by kfir-azu — CHANGES_REQUESTED** [review]

File: src/assets/components/shared/Button/Button.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import CircularProgress from '@mui/material/CircularProgress';
+import { getCircularSize, sizeDimensions, StyledButton, StyledButtonProps } from "./styles";
+import { ButtonSize } from '../../types';
+
+
+
+export interface ButtonProps extends StyledButtonProps{
```
[the remark this answers is not in the document]
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:7]: done

File: src/assets/components/shared/Button/Button.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  onClick: () => void;
+}
+
+export default function Button({
+  $variant,
```
[the remark this answers is not in the document]
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:14]: done

File: src/assets/components/shared/Button/Button.tsx [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+  
+
+  return (
+    <StyledButton
+    $variant={$variant}
+      $size={$size}
+      onClick={onClick}
+      disabled={isDisabled}
+    >
+      {(startIcon && !isLoading) && startIcon}
```
[the remark this answers is not in the document]
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/Button/Button.tsx:32]: done

**Review by kfir-azu — APPROVED** [review]

**Pull request merged**